platform/src/main/java/org/hillview/sketches/Histogram.java

```diff
@@ -32,6 +33,35 @@ public Histogram(final IHistogramBuckets bucketDescription) {
         this.buckets = new long[bucketDescription.getNumOfBuckets()];
     }
 
+    // Note that we ceil in order to round up to the nearest leaf
+    public static int intLog2(int x) {
+        return (int)(Math.ceil(Math.log(x) / Math.log(2)));
+    }
+
+    // The amount of noise depends on the number of "leaves" that are required to compute the interval.
+    public int noiseMultiplier(int bucketIdx) {
+        // Cast down - the noise operation is only valid on the dyadic histogram leaves
```

Comment:
If you subclass Histogram then this can be made slightly safer too by storing a pointer already cast.